crates/libtiny_tui/src/msg_area/mod.rs

```diff
@@ -179,11 +179,14 @@ impl MsgArea {
     }
 
     /// Recalculate the scroll offset due to resizing of the window
-    fn recalculate_scroll(&mut self, old_height: i32, old_total_lines: i32) {
-        if self.scroll > 0 {
-            let ratio = (self.scroll as f32 + old_height as f32) / old_total_lines as f32;
-            let total_lines = self.update_total_visible_lines();
-            self.scroll = max(0, ((ratio * total_lines as f32) as i32) - self.height);
+    fn recalculate_scroll(&mut self, old_height: i32, old_total_lines: Option<i32>) {
```

Comment:
This function cannot handle `None` case, so if you keep the type as before it will become evident from the type that this really needs a `old_total_lines` argument. Please revert this and call this function only when `old_total_line` is available. It will be less surprising when you look at this code in a few months.

Reply:
> So we could reset total_height as Some(0) in clear, which is probably something we should do regardless of this bug, but it should also fix the bug.

This is right.  Sorry